--- a/scripts/Makefile.build
+++ b/scripts/Makefile.build
@@ -258,7 +258,7 @@ quiet_cmd_link_o_target = LD      $@
 # If the list of objects to link is empty, just create an empty built-in.o
 # -nostdlib is added to make "make LD=gcc ..." work (some people use that)
 cmd_link_o_target = $(if $(strip $(obj-y)),\
-		$(LD) -nostdlib $(ld_flags) -r -o $@ $(filter $(obj-y), $^),\
-		rm -f $@; $(AR) rcs $@)
+		$(LD) -nostdlib $(ld_flags) -r -o $@ $(filter $(obj-y), $^),\
+		rm -f $@; touch $@)
 
 $(builtin-target): $(obj-y) FORCE
@@ -271,7 +271,7 @@ $(subdir-builtin): $(obj)/%/built-in.o: $(obj)/% ;
 # Rule to compile a set of .o files into one .o file
 #
 quiet_cmd_link_l_target = AR      $@
-cmd_link_l_target = rm -f $@; $(AR) $(EXTRA_ARFLAGS) rcs $@ $(lib-y)
+cmd_link_l_target = rm -f $@; if [ -n "$(lib-y)" ]; then $(AR) $(EXTRA_ARFLAGS) rcs $@ $(lib-y); else touch $@; fi
 
 $(lib-target): $(lib-y) FORCE
 	$(call if_changed,link_l_target)

